inngest/src/main/kotlin/com/inngest/InngestFunctionConfigBuilder.kt

```diff
@@ -127,6 +128,25 @@ class InngestFunctionConfigBuilder {
      */
     fun retries(attempts: Int): InngestFunctionConfigBuilder = apply { this.retries = attempts }
 
+    /**
+     * Configure function throttle limit
+     *
+     * @param limit The total number of runs allowed to start within the given period. The limit is applied evenly over the period.
+     * @param period The period of time for the rate limit. Run starts are evenly spaced through the given period.
+     * The minimum granularity is 1 second.
+     * @param burst The number of runs allowed to start in the given window in a single burst.
+     * A burst > 1 bypasses smoothing for the burst and allows many runs to start at once, if desired. Defaults to 1, which disables bursting.
+     * @param key An optional expression which returns a throttling key for controlling throttling.
+     * Every unique key is its own throttle limit. Event data may be used within this expression, eg "event.data.user_id".
+     */
+    @JvmOverloads
+    fun throttle(
+        limit: Int,
+        period: Duration,
+        burst: Int? = null,
```

Comment:
I should have mentioned it in the description but `rateLimit` and `throttle` are different configs, the former being lossy as in if the limit is hit function runs are skipped: https://github.com/inngest/inngest-js/blob/809b4efec259a608ce77a004d98fbc2f36d2bc3a/packages/inngest/src/components/InngestFunction.ts#L365-L420

I believe `throttle` is missing completely from the spec. We do have a ticket for `rateLimit` as well: https://linear.app/inngest/issue/INN-3332/add-ratelimit-configuration

Reply:
>would this default value of null overwrite the default of 1 in the data class? I guess it doesn't technically matter if it doesn't get serialized to the Inngest server and if they default to 1 there

Good catch, honestly i have no idea so it's better to avoid the confusion and initialize it to `null`. The JS SDK skips it when it's null so yeah it's up to the server.
###### Request
```bash
POST /fn/register HTTP/1.1
host: 127.0.0.1:8288
connection: keep-alive
Content-Type: application/json
User-Agent: inngest-js:v3.22.5
x-inngest-sdk: inngest-js:v3.22.5
x-inngest-framework: nextjs
x-inngest-expected-server-kind: dev
Server-Timing: handler
Authorization: Bearer undefined
accept: */*
accept-language: *
sec-fetch-mode: cors
accept-encoding: gzip, deflate
content-length: 730

{"url":"http://localhost:3000/api/inngest","deployType":"ping","framework":"nextjs","appName":"new-app","functions":[{"id":"new-app-hello-world","name":"hello-world","triggers":[{"event":"hello-world"}],"steps":{"step":{"id":"step","name":"step","runtime":{"type":"http","url":"http://localhost:3000/api/inngest?fnId=new-app-hello-world&stepId=step"}}},"throttle":{"limit":1,"period":"1s"}},{"id":"new-app-failing-function","name":"failing-function","triggers":[{"event":"fn-fail"}],"steps":{"step":{"id":"step","name":"step","runtime":{"type":"http","url":"http://localhost:3000/api/inngest?fnId=new-app-failing-function&stepId=step"},"retries":{"attempts":0}}}}],"sdk":"js:v3.22.5","v":"0.1","capabilities":{"trust_probe":"v1"}}
```
###### Response
```bash
HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
Vary: Origin
X-Inngest-Server-Kind: dev
Date: Sun, 01 Sep 2024 15:20:05 GMT
Content-Length: 1273

{
  "version": "0.29.6-2e911abc6",
  "authed": false,
  "startOpts": {
    "dir": "",
    "urls": [
      "http://127.0.0.1:8080/api/inngest"
    ],
    "autodiscover": true,
    "poll": true,
    "poll_interval": 5,
    "tick": 150000000,
    "retry_interval": 0
  },
  "functions": [
    {
      "id": "6e55e21b-b344-56bd-8fdf-233a81768e3a",
      "fv": 0,
      "name": "hello-world",
      "slug": "new-app-hello-world",
      "triggers": [
        {
          "event": "hello-world"
        }
      ],
      "throttle": {
        "burst": 1,
        "limit": 1,
        "period": "1s"
      },
      "steps": [
        {
          "id": "step",
          "name": "step",
          "uri": "http://localhost:3000/api/inngest?fnId=new-app-hello-world\u0026stepId=step"
        }
      ]
    },
    {
      "id": "9001ee96-8157-5588-b51e-381438392f9e",
      "fv": 0,
      "name": "failing-function",
      "slug": "new-app-failing-function",
      "triggers": [
        {
          "event": "fn-fail"
        }
      ],
      "steps": [
        {
          "id": "step",
          "name": "step",
          "uri": "http://localhost:3000/api/inngest?fnId=new-app-failing-function\u0026stepId=step",
          "retries": 0
        }
      ]
    }
  ],
  "handlers": null
}
```

Reply:
cool thanks for checking! totally missed we have another ticket for rateLimit